tests/test_api.py

```diff
@@ -113,48 +128,81 @@ def test_no_memberships(self, client):
         results = response.json()['results']
         assert results == []
 
-    def test_list_memberships(self, client):
-        MembershipFactory.create_batch(3, group=CourseAccessGroupFactory.create())
+    @pytest.mark.parametrize('org_name, status_code, expected_count', [
+        ['my_org', 200, 3],
+        ['other_org', 200, 0],
+    ])
+    def test_list_memberships(self, client, org_name, status_code, expected_count):
+        org = Organization.objects.get(name=org_name)
+        MembershipFactory.create_batch(3, group__organization=org)
         response = client.get(self.url)
-        assert response.status_code == 200
         results = response.json()['results']
-        assert len(results) == 3
-
-    def test_one_membership(self, client):
-        membership = MembershipFactory.create()
+        assert response.status_code == status_code, response.content
+        assert len(results) == expected_count
+
+    @pytest.mark.parametrize('group_org, status_code, skip_response_check', [
+        ['my_org', 200, False],
+        ['other_org', 404, True],
+    ])
+    def test_one_membership(self, client, group_org, status_code, skip_response_check):
+        org = Organization.objects.get(name=group_org)
+        membership = MembershipFactory.create(group__organization=org)
         response = client.get('/memberships/{}/'.format(membership.id))
+        assert response.status_code == status_code, response.content
         result = response.json()
-        assert result == {
+        assert skip_response_check or (result == {
             'id': membership.id,
-            'user': membership.user.id,
-            'user_email': membership.user.email,
-            'user_username': membership.user.username,
-            'group': membership.group.id,
-            'group_name': membership.group.name,
-            'group_description': membership.group.description,
-        }
-
-    def test_add_membership(self, client):
+            'user': {
+                'id': membership.user.id,
+                'email': membership.user.email,
+                'username': membership.user.username,
+            },
+            'group': {
+                'id': membership.group.id,
+                'name': membership.group.name,
+            },
+        }), 'Verify the serializer results.'
+
+    @pytest.mark.parametrize('group_org, user_org, status_code, expected_count, check_new_membership', [
+        ['my_org', 'my_org', 201, 1, True],  # Should work for own users and groups
```

Comment:
@OmarIthawi What do you think of using DRF status code constants? I know we all know the codes, but a junior engineer might not and using `status.HTTP_BAD_REQUEST` could be clearer than `400`

Ref: https://www.django-rest-framework.org/api-guide/status-codes/#successful-2xx

Reply:
Yeah, I think magic constants aren't good. I'll do that, thanks @johnbaldwin!